actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py

```diff
@@ -0,0 +1,294 @@
+        overwrite: bool,
+        metadata: str,
+    ) -> Union[DiagnosticSheetResult, DiagnosticSheetError]:
+        # Don't use dask for image reading
+        aicsimageio.use_dask(False)
+
+        try:
+            # Get the ultimate end save paths for grouped plot
+            if row["CellImage2DAllProjectionsPath"]:
+                if row[str(metadata)]:
+                    diagnostic_sheet_save_path = (
+                        diagnostic_sheet_dir / f"{metadata}_{row[metadata]}.png"
+                    )
+                    log.info(f"Generating diagnostic sheet for cell ID: {row.CellId}")
+                else:
+                    # else no path to save
+                    diagnostic_sheet_save_path = None
+
+            # Check skip
+            if not overwrite and diagnostic_sheet_save_path.is_file():
+                log.info(f"Skipping diagnostic sheet for Cell Id: {row.CellId}")
+                return DiagnosticSheetResult(row.CellId, None)
+
+            # Return ready to save image
+            return DiagnosticSheetResult(row.CellId, diagnostic_sheet_save_path)
+        # Catch and return error
+        except Exception as e:
+            log.info(
+                f"Failed to retrieve the CellImage2DAllProjectionsPath"
+                f"for cell ID: {row.CellId}"
+                f"Error: {e}"
+            )
+            return DiagnosticSheetError(row.CellId, str(e))
+
+    @log_run_params
+    def run(
+        self,
+        dataset: Union[str, Path, pd.DataFrame, dd.DataFrame],
+        distributed_executor_address: Optional[str] = None,
+        overwrite: bool = False,
+        metadata: Optional[str] = None,
+        feature: Optional[str] = None,
+        **kwargs,
+    ):
+        """
+        Run a pure function.
+
+        Protected Parameters
+        --------------------
+        distributed_executor_address: Optional[str]
+            An optional executor address to pass to some computation engine.
+        clean: bool
+            Should the local staging directory be cleaned prior to this run.
+            Default: False (Do not clean)
+        debug: bool
+            A debug flag for the developer to use to manipulate how much data runs,
+            how it is processed, etc.
+            Default: False (Do not debug)
+
+        Parameters
+        ----------
+        dataset: Union[str, Path, pd.DataFrame, dd.DataFrame]
+            The primary cell dataset to use for generating 
+            diagnistic sheet for a group of cells.
+
+            **Required dataset columns:** *["CellId"]*
+
+        metadata: str
+            The metadata to group cells and generate a diagnostic sheet. 
+            For example, "FOVId"
+
+        feature: str
+            The name of the single cell feature to display. For example, "imsize_orig"
+
+        distributed_executor_address: Optional[str]
+            An optional executor address to pass to some computation engine.
+            Default: None
```

Comment:
Duplicated from above "protected parameters" when you merge those two sections make sure to remove this duplicate.